src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs

```diff
@@ -164,6 +179,26 @@ private static void GetTempPath(ref ValueStringBuilder builder)
                 throw Win32Marshal.GetExceptionForLastWin32Error();
 
             builder.Length = (int)result;
+
+            static uint GetTempPathW(int bufferLen, ref char buffer)
+            {
+                delegate* unmanaged<int, char*, uint> func = s_GetTempPathWFunc;
```

Comment:
@jkotas, I assume we don't need to be concerned about introduced reads such that `s_GetTempPathWFunc` would need to be volatile?

Reply:
This can use `volatile` on the variable for a good measure.

It should work without it today, but I think it is a good practice to use `volatile` for anything lock-free by default.